Managers/AudioMan.cpp

```diff
@@ -1032,21 +1032,25 @@ bool AudioMan::PlaySound(Sound *pSound, int priority, float distance, double pit
 
 	// Set sample's channel looping setting
 	ga_Handle * handle;
-	if (pSound->m_Loops > 0)
+	if (pSound->m_Loops == 0)
+	{
+		handle = gau_create_handle_sound(m_pMixer, pSound->GetCurrentSample(), 0, 0, 0);
+	}
+	else
 	{
 		gau_SampleSourceLoop* loopSrc; 
 		handle = gau_create_handle_sound(m_pMixer, pSound->GetCurrentSample(), 0, 0, &loopSrc);
 		gau_sample_source_loop_set(loopSrc, -1, 0);
 	}
-	else 
-	{
-		handle = gau_create_handle_sound(m_pMixer, pSound->GetCurrentSample(), 0, 0, 0);
-	}
+
 	m_SoundChannels[channel] = handle;
 	m_SoundInstances[channel] = pSound->GetCurrentSample();
 
 	ga_handle_play(handle);
 
+	// Due to Gorilla lacking the ability to set a master volume, we have to set it here.
+	ga_handle_setParamf(handle, GA_HANDLE_PARAM_GAIN, m_SoundsVolume);
```

Comment:
Hmm, after some looking through I think I see what the cause of the problem is (good work fixing it) - namely that when a sound channel is chosen, it's destroyed first to clear all its data, which means that SetSoundsVolume on line 320 gets overwritten. I don't think there's a reasonable alternative to this, since I'd guess we need to destroy the sound handle for memory reasons, so instead I have a few requests to make this cleaner:
1. Given it does nothing, why not remove SetSoundsVolume and all calls to it, or at least put a comment or something marking it as useless so we can clean it in future.
2. Make this setParamf you've added mimic the one in SetSoundsVolume so it sets the sound volume to MAX_VOLUME * m_SoundsVolume. Since MAX_VOLUME is defined as 1.0 this won't change things, but it's consistent with expectations so I think it's worth fixing.
3. I don't really like this setting where you've put it; I'd say that, given it's an initial configuration for the sound handle, it should be as early as reasonably possible. So this means either line 1045 (i.e. after creating it but before adding it to m_SoundChannels) or 1048 (i.e. after adding it to m_SoundChannels but before playing it) in your code I guess. I know this is also very minor, but it feels conceptually weird to change the volume after the sound has been played.